### PR TITLE
snap,snappy: allow snaps to require system features

### PR DESCRIPTION
**Pull request by niemeyer**

### CODE REVIEW

File: snappy/overlord.go [hunk 1]
```diff
@@ -53,6 +54,10 @@ func CheckSnap(snapFilePath string, curInfo *snap.Info, flags InstallFlags, mete
 		return err
 	}
 
+	if len(s.Assumes) > 0 {
```
Comment (mvo5) [inline snappy/overlord.go]: Heh, very nice!

**Comment by mvo5** [comment]
Looks good. The key `assumes` is fine, I was also thinking of `needs` or `wants` as potential names (because assumes has a slightly less strong ring to it for me). But that is bike-shedding, so :+1:

**Comment by niemeyer** [comment]
From our chat:

Gustavo Niemeyer, [15.04.16 04:43]
So, assumes.. the idea for the name was to avoid the concept of needs/consumes/requires/depends..
as that's generally taken to mean something else which we're already digging into slowly.

We'll likely have some sort of "requires" field soon, talking about plugs.. "requires these to be established", which to some degree becomes a sort of old-school Depends. "assumes" is also interesting because it implies the snap is actually assuming something works, instead of taking measures for when it doesn't. So the blocking is somewhat implied.

Michael Vogt, [15.04.16 04:46]
thanks, its a bit unusual but that is probably actually a plus. so +1

Gustavo Niemeyer, [15.04.16 04:47]
Indeed

**Comment by zyga** [comment]
Indeed, looks interesting. Our own feature flags :)

**Pull request merged**